daemon/mgr/container.go

```diff
@@ -1302,6 +1305,26 @@ func (mgr *ContainerManager) Resize(ctx context.Context, name string, opts types
 	return mgr.Client.ResizeContainer(ctx, c.ID, opts)
 }
 
+// Wait stops processing until the given container is stopped.
+func (mgr *ContainerManager) Wait(ctx context.Context, name string) (types.ContainerWaitOKBody, error) {
+	c, err := mgr.container(name)
+	if err != nil {
+		return types.ContainerWaitOKBody{}, err
+	}
+
+	// We should notice that container's meta data shouldn't be locked in wait process, otherwise waiting for
+	// a running container to stop would make other client commands which manage this container are blocked.
+	// If a container status is exited or stopped, return exit code immediately.
+	if c.IsExited() || c.IsStopped() {
```

Comment:
How about the `created` container? What is the behaviour?

Reply:
Yeah, the `created` case is very special, this concern is very considerate. So I dive into the code and find that when a new container is created and *`running`*, this map `watch.containers` would record container's pack info. If a container state is created, this code won't go to if statement codeblock and will run to `WaitContainer` function in `ctrd/container.go`, so this `WaitContainer` function will call `ProbeContainer` function, and `ProbeContainer` function will [look up](https://github.com/alibaba/pouch/blob/master/ctrd/container.go#L144) container pack info from the above map `watch.containers`. This function [`watch.notify()`](https://github.com/alibaba/pouch/blob/master/ctrd/watch.go#L140) will look up map `watch.containers`, so when the container is just created, the look up process will [fail](https://github.com/alibaba/pouch/blob/master/ctrd/watch.go#L144) and just return the `Message{}` to channel `ch`, so the `ProbeContainer` function will get `Message{}` channel struct and this struct's member `uint 32 ExitCode` would be initialized with default number `0`. So when waiting a created container, the client will see the code `0`, which is the same as moby.

Reply:
So, can we add the case for this? Thanks